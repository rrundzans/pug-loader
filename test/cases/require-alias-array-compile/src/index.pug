img(src=require('./image.jpeg'))

- var basename = 'logo';

//- test the prefix only as alias
img.prefix(src=require('@/images/logo.jpeg'))

//- test the single char as alias
img.char(src=require('&/images/logo.jpeg'))

//- note: if alias value is a string, the variable can be used

//- test alias value as string
img.a1(src=require('ImagesA/' + basename + '-a1.jpeg'))
img.a2(src=require('~ImagesA/' + basename + '-a2.jpeg'))
img.a3(src=require('@ImagesA/' + basename + '-a3.jpeg'))

//- test alias value as string contained a prefix in name
img.b1(src=require('ImagesB/logo-b1.jpeg'))
img.b2(src=require('~ImagesB/logo-b2.jpeg'))
img.b3(src=require('@ImagesB/logo-b3.jpeg'))
//- this works too
-
  var logos = {
    bc1: require('ImagesB/logo-b1.jpeg'),
    bc2: require('~ImagesB/logo-b2.jpeg'),
    bc3: require('@ImagesB/logo-b3.jpeg'),
  };
each src, className in logos
  img(src=src class=className)

//- test alias value as array
//- LIMITATION for compile method:
    If the alias is an array of paths, then only the first path is returned
    because it is not possible to determine the correct path containing the interpolated filename.
//-img.c1(src=require('ImagesC/' + basename + '-c1.jpeg'))
img.c1(src=require('ImagesC/logo-c1.jpeg'))
img.c2(src=require('~ImagesC/logo-c2.jpeg'))
img.c3(src=require('@ImagesC/logo-c3.jpeg'))

//- test alias value as array contained a prefix in name
img.d1(src=require('ImagesD/logo-d1.jpeg'))
img.d2(src=require('~ImagesD/logo-d2.jpeg'))
img.d3(src=require('@ImagesD/logo-d3.jpeg'))