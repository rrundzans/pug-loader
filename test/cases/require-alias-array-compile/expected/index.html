<img src="/assets/images/image.f78b30f4.jpeg"><img class="prefix" src="/assets/images/logo.f78b30f4.jpeg"><img class="char" src="/assets/images/logo.f78b30f4.jpeg"><img class="a1" src="/assets/images/logo-a1.f78b30f4.jpeg"><img class="a2" src="/assets/images/logo-a2.f78b30f4.jpeg"><img class="a3" src="/assets/images/logo-a3.f78b30f4.jpeg"><img class="b1" src="/assets/images/logo-b1.f78b30f4.jpeg"><img class="b2" src="/assets/images/logo-b2.f78b30f4.jpeg"><img class="b3" src="/assets/images/logo-b3.f78b30f4.jpeg"><img class="bc1" src="/assets/images/logo-b1.f78b30f4.jpeg"><img class="bc2" src="/assets/images/logo-b2.f78b30f4.jpeg"><img class="bc3" src="/assets/images/logo-b3.f78b30f4.jpeg"><img class="c1" src="/assets/images/logo-c1.f78b30f4.jpeg"><img class="c2" src="/assets/images/logo-c2.f78b30f4.jpeg"><img class="c3" src="/assets/images/logo-c3.f78b30f4.jpeg"><img class="d1" src="/assets/images/logo-d1.f78b30f4.jpeg"><img class="d2" src="/assets/images/logo-d2.f78b30f4.jpeg"><img class="d3" src="/assets/images/logo-d3.f78b30f4.jpeg">